packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts

```diff
@@ -22,24 +23,41 @@ export const useVisualizedDataStreams = (queries: TimeSeriesDataQuery[], passedI
     },
   });
 
-  // Line | Scatter | Bar charts do not support alarm streams.
-  const dataStreamsWithoutAlarms = dataStreams.filter(isNotAlarmStream);
-
-  const hasError = useMemo(() => dataStreamsWithoutAlarms.some(dataStreamHasError), [dataStreamsWithoutAlarms]);
-
-  const isLoading = useMemo(
-    () => !hasError && dataStreamsWithoutAlarms.some(dataStreamIsLoading),
-    [hasError, dataStreamsWithoutAlarms]
-  );
-
-  const visibleData = dataStreamsWithoutAlarms.flatMap(({ data }) => getVisibleData(data, utilizedViewport, false));
+  // dependencies
+  const dataString = JSON.stringify(dataStreams);
+  const thresholdsString = JSON.stringify(thresholds);
+  const utilizedString = JSON.stringify(utilizedViewport);
+
+  // utilizing state to make sure the values are updated only on change
+  const [dataStreamsWithoutAlarms, setDataStreamsWithoutAlarms] = useState(dataStreams);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [performanceMode, setPerformanceMode] = useState(false);
+
+  useEffect(() => {
+    const dataWOAlarms = dataStreams.filter(isNotAlarmStream) as DataStream[];
+    setDataStreamsWithoutAlarms(dataWOAlarms);
+
+    const hasErrorLocal = dataWOAlarms.some(dataStreamHasError);
+    const isLoadingLocal = !hasErrorLocal && dataWOAlarms.some(dataStreamIsLoading);
+    setHasError(hasErrorLocal);
+    setIsLoading(isLoadingLocal);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataString, thresholdsString]);
+
+  useEffect(() => {
+    const visibleData = dataStreamsWithoutAlarms.flatMap(({ data }) => getVisibleData(data, utilizedViewport, false));
+    const performanceMode1 = visibleData.length > PERFORMANCE_MODE_THRESHOLD;
```

Comment:
can we have a better name than `performanceMode1`

Reply:
gahhh! my bad